src/bluesearch/database/identifiers.py

```diff
@@ -0,0 +1,80 @@
+"""Module for handling identifiers."""
+
+# Blue Brain Search is a text mining toolbox focused on scientific use cases.
+#
+# Copyright (C) 2020  Blue Brain Project, EPFL.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import hashlib
+from typing import List, Tuple
+
+import pandas as pd
+
+
+def generate_uid(identifiers: Tuple) -> str:
+    """Generate a deterministic UID for the given identifiers.
+
+    Parameters
+    ----------
+    identifiers
+        Values of the identifiers.
+
+    Returns
+    -------
+    int
+        A deterministic UID.
+    """
+    data = str(identifiers).encode()
+    hashed = hashlib.md5(data).hexdigest()  # nosec
+    return hashed
+
+
+def generate_uids(metadata: pd.DataFrame, identifiers: List[str]) -> pd.DataFrame:
```

Comment:
I'm slightly concerned by seeing a dataframe taken as an argument this way. It could have any arbitrary structure and values and it's not really obvious what this dataframe is expected to contain. I think one could justify this if we had a very precise spec for the dataframe that we use to hold our data, but I don't think we have one. And why do we expect the identifier data to be in dataframe anyway?

So if anything, I would prefer a more structured data format like a sequence of tuples or something like that.  Actually then it's just about mapping `generate_uid` across that sequence, which is pretty trivial, so maybe this function is superfluous and we can just call `generate_uid` wherever necessary straight away?

Reply:
> And why do we expect the identifier data to be in dataframe anyway?

Using a DataFrame is what is done at the moment in our code: https://github.com/BlueBrain/Search/blob/93c13761f16b16a85b5b510b61f87ad06de752b3/src/bluesearch/database/cord_19.py#L123.

> we can just call generate_uid wherever necessary straight away?

That's a possibility. However, one needs to remember how to call `generate_uid(...)` and would be responsible to guarantee that there is no paper without identifiers (see https://github.com/BlueBrain/Search/pull/418#discussion_r707135976 for an explanation).

Reply:
Just a note here that `metadata.csv` is specific to the CORD-19.

Reply:
`generate_uids(...)` removed in 26bea59.

Reply:
Thanks for addressing my comment!